### PR TITLE
New r.fill.category module

### PR DESCRIPTION
**Pull request by pzatelli**
r.replace substitutes the values of pixels of a given category with values of the surrounding pixels and stores the output in a new raster map. Pixels with other categories are left untouched.
It combines r.neighbors and r.mapcalc. Optionally, it uses r.out.mpeg to create animations.
See the html manual page for an example application on the North Carolina sample dataset.

An application of this module will be presented at FOSS4G 2019 Bucharest.

The module has been prepared according to https://trac.osgeo.org/grass/wiki/Submitting and https://trac.osgeo.org/grass/wiki/Submitting/Python

### CODE REVIEW

File: grass7/raster/r.replace/r.replace.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+#
+#   This program is free software under the GNU General Public
+#   License (>=v2). Read the file COPYING that comes with GRASS
+#   for details.
+#
+#############################################################################
+#%Module
+#% description: Replaces the values of pixels of a given category with values of the surrounding pixels.
+#% keyword: raster
+#% keyword: replace
```
Comment (neteler) [inline grass7/raster/r.replace/r.replace.py]: Please use "algebra" as the second keyword (the "topics" manual page [1] is generated from the second keyword)

[1] https://grass.osgeo.org/grass77/manuals/topic_algebra.html

**Comment by neteler** [comment]
Just FYI - here a current case in Chile of a disappearing lake:
https://twitter.com/SERTIT_RM/status/1113089369521434630?s=19

**Comment by pzatelli** [comment]
Sorry for the delay, we propose the new module name r.fill.category.
An add-on named r.fill.gaps already exists and it performs a similar task when using mode=mode, but the outcome is different, mainly because it does not iterate. It also works on NULL values, but obviously it is possibile to use r.null to set the category to fill to NULL.
If the name is acceptable I can update my repository fork, I am not sure this will update the current pull request or will result in a new pull request because the name of the directory and of the files are different.

**Comment by petrasovaa** [comment]
r.fill.category is ok I think. Unfortunately I don't know if the updating will work as expected, but I think it could.

**Comment by neteler** [comment]
(I have taken liberty to also change the PR title from "r.replace" -> "r.fill.category" - just to avoid confusion)

**Comment by neteler** [comment]
@petrasovaa , @pzatelli - is anything left open here? Time to merge?

**Comment by pzatelli** [comment]
Unless further changes are required, it's ok to merge for me.

**Pull request merged**